content/setup/local-with-vagrant/_index.md

```diff
@@ -0,0 +1,198 @@
++++
+title = "Local with Vagrant"
+date = 2020-06-08T16:16:15+05:30
+draft = false
+weight = 20
+toc = true
++++
+
+An easy way to try out Tinkerbell locally is via [Vagrant](https://www.vagrantup.com).
+
+Vagrant wraps and glues together different technologies to help you write as
+code repetitive environment mixing virtualization or cloud providers such as
+VirtualBox, libvirt, AWS, GCP with provisioning techniques like Puppet, Ansible,
+Bash.
+
+In this context we use VirtualBox or libvirt as a virtualization platform and
+bash for provisioning all the required tools and configurations.
+
+Following this tutorial, you will end up with a provisioner up and running and a
+worker ready to handle actions and workflows.
+
+## Prerequisites
+
+- [The host's processor should support virtualization](https://www.cyberciti.biz/faq/linux-xen-vmware-kvm-intel-vt-amd-v-support/)
+- [Vagrant](https://www.vagrantup.com/downloads) is installed
+- Either [VirtualBox](https://www.virtualbox.org/) or [libvirtd](https://libvirt.org/) is installed and configured.
+
+## Cloning Tinkerbell
+
+1. Clone `tink` and move inside `deploy/vagrant` directory. This folder contains
+   vagrant configuration (Vagrantfile) and the scripts needed to provide workers
+   and provisioners.
+
+```
+$ git clone https://github.com/tinkerbell/tink.git
+$ cd tink/deploy/vagrant
+```
+
+## Start the provisioner
+
+```
+$ vagrant up provisioner
+Bringing machine 'provisioner' up with 'virtualbox' provider...
+
+    ......
+
+    INFO: tinkerbell stack setup completed successfully on ubuntu server
+    NEXT: 1. Enter ./deploy and run: source ../envrc; docker-compose up
+           2. Try executing your fist workflow.
+              Follow the steps described in https://tinkerbell.org/examples/hello-world/ to say 'Hello World!' with a workflow.
+```
+
+When the provisioner is ready, a summary is printed to the console and you can now ssh in.
+
+### Connecting to the Provisioner
+
+All the following commands have to be executed in the artifacts' directory created above unless stated otherwise.
+
+Connect to the provisioner via `vagrant ssh`:
+
+```
+$ vagrant ssh provisioner
+vagrant@provisioner:~$
+```
+
+At this point, you are in an Ubuntu box that has a couple of utils installed.
+
+Let's now start up the Tinkerbell stack with `docker-compose`:
+
+```
+$ sudo su -l
+$ cd /vagrant && source envrc && cd deploy
```

Comment:
Since you ran `sudo`. Now prefix commands with `#` to show we're running as root?

```
vagrant@provisioner:~$ sudo su -l
root@provisioner:~# cd /vagrant && source envrc && cd deploy
root@provisioner:/vagrant/deploy# docker-compose up -d
```

Reply:
This is not required as sudo anymore!! 👍 didn't know about the `#` thing at all.
Thanks